app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -66,30 +40,63 @@ const ClientEligibilities = () => {
   };
 
   const toggleChecked = (eligibilityGroup: clientEligibilityGroup, eligibilityIndex: number) => {
-    const updatedEligibility = {
-      ...eligibilityGroup.eligibilities[eligibilityIndex],
-      checked: !eligibilityGroup.eligibilities[eligibilityIndex].checked,
-    };
-
-    const updatedEligibilities = [
-      ...eligibilityGroup.eligibilities.slice(0, eligibilityIndex),
-      updatedEligibility,
-      ...eligibilityGroup.eligibilities.slice(eligibilityIndex + 1),
-    ];
+    const targetEligibility = eligibilityGroup.eligibilities[eligibilityIndex];
+    const targetToggleState = !targetEligibility.checked;
+    let updatedEligibilities;
+
+    // If a user toggles "See all", we (un)check all of the eligibilities accordingly
+    if (targetEligibility.name === 'See all') {
+      updatedEligibilities = massToggleGroupEligibilities(
+        eligibilityGroup.eligibilities,
+        targetToggleState,
+      );
+    } else {
+      // If a user untoggles a specific eligiblity, this code also untoggles "See all"
+      // Currently, there is no logic to toggle "See all" if a user toggles all individual
+      // eligibilities given that this is somewhat of an implicit understanding; however, this
+      // can be added in the future
```

Comment:
If we get around to refactoring this in the future, one design we could consider is to not represent "See all" in the React state at all and instead model it as a function of the states of the other checkboxes. It's an application of the concept of "[Making illegal states unrepresentable](https://v5.chriskrycho.com/journal/making-illegal-states-unrepresentable-in-ts/)", where it would be impossible to represent the state of having the "See all" checkbox unchecked at the same time as when all the other checkboxes in the group are checked, and I think it would clean up this toggle handler function, since you wouldn't have to special case these edge cases for ensuring the state is consistent.

Reply:
Great, thanks for sharing that link. I bookmarked it to read during some down time and can refactor in the future.